packages/@aws-cdk/aws-s3-deployment/lib/bucket-deployment.ts

```diff
@@ -193,18 +203,53 @@ export class BucketDeployment extends CoreConstruct {
       throw new Error('Distribution must be specified if distribution paths are specified');
     }
 
+    if (props.enableTempEfsStorage && !props.vpc) {
+      throw new Error('Vpc must be specified if enableTempEfsStorage is true');
+    }
+
+    const ACCESS_POINT_PATH = '/lambda';
+    const ACCESS_MODE = '0777';
+    let accessPoint;
+    if (props.enableTempEfsStorage && props.vpc) {
+      const fileSystem = new efs.FileSystem(this, 'FileSystem', {
+        vpc: props.vpc,
+        removalPolicy: cdk.RemovalPolicy.DESTROY,
+      });
+      accessPoint = fileSystem.addAccessPoint('AccessPoint', {
+        path: ACCESS_POINT_PATH,
+        createAcl: {
+          ownerUid: '1001',
+          ownerGid: '1001',
+          permissions: ACCESS_MODE,
+        },
+        posixUser: {
+          uid: '1001',
+          gid: '1001',
+        },
+      });
+    }
+
+    const MOUNT_PATH = `/mnt${ACCESS_POINT_PATH}`;
     const handler = new lambda.SingletonFunction(this, 'CustomResourceHandler', {
       uuid: this.renderSingletonUuid(props.memoryLimit),
       code: lambda.Code.fromAsset(path.join(__dirname, 'lambda')),
       layers: [new AwsCliLayer(this, 'AwsCliLayer')],
       runtime: lambda.Runtime.PYTHON_3_6,
+      environment: props.enableTempEfsStorage ? {
+        IS_EFS_ENABLED: 'true',
+        MOUNT_PATH: MOUNT_PATH,
+      } : undefined,
       handler: 'index.handler',
       lambdaPurpose: 'Custom::CDKBucketDeployment',
       timeout: cdk.Duration.minutes(15),
       role: props.role,
       memorySize: props.memoryLimit,
       vpc: props.vpc,
       vpcSubnets: props.vpcSubnets,
+      filesystem: accessPoint ? lambda.FileSystem.fromEfsAccessPoint(
```

Comment:
Each lambda invocation creates its own `/uuid` directory under `/mnt/lambda`. Therefore each invocation or lambda config will have its own EFS path and there won't be any conflicts. 
Also, i ran a test where i created two `BucketDeployment` objects and it created two different EFS volume for both the objects.

Reply:
Even though it created two EFS volumes, there was only 1 lambda function in the stack which didn't have EFS env variables. Looking further into it.

Reply:
Updated `renderSingletonUuid` as below

```ts
  private renderSingletonUuid(memoryLimit?: number, useEfs?: boolean) {
    let uuid = '8693BB64-9689-44B6-9AAF-B0CC9EB8756C';

    // if user specify a custom memory limit, define another singleton handler
    // with this configuration. otherwise, it won't be possible to use multiple
    // configurations since we have a singleton.
    if (memoryLimit) {
      if (cdk.Token.isUnresolved(memoryLimit)) {
        throw new Error('Can\'t use tokens when specifying "memoryLimit" since we use it to identify the singleton custom resource handler');
      }

      uuid += `-${memoryLimit.toString()}MiB`;
    }

    // if user specify to use Efs storage, define another singleton handler
    // with this configuration. otherwise, it won't be possible to use multiple
    // configurations since we have a singleton.
    if (useEfs) {
      if (cdk.Token.isUnresolved(useEfs)) {
        throw new Error('Can\'t use tokens when specifying "useEfs" since we use it to identify the singleton custom resource handler');
      }
      uuid += '-EfsEnabled';
    }

    return uuid;
  }
```

It generated two different lambda function upon creating two `BucketDeployment` objects.